src/index.ts

```diff
@@ -292,30 +219,30 @@ export const plugin: PluginFunction = async (
     const properties = toSwiftProperties(op.selections, fragments, scalarMap, selectionSets);
 
     return `
-class ${op.name} {
+class ${op.name}: Decodable {
   let data: Data?;
   let errors: Errors?;
   public static let operationDefinition: String =
     """
 ${indentMultiline(op.definition, 2)}
     """
 
-  init(json: [String: Any]) {
-    self.data = Data(json: json["data"] as Any)
-    self.errors = Errors(json: json["errors"] as Any)
+  private enum CodingKeys: String, CodingKey {
+    case data
+    case errors
   }
 
-${indentMultiline(printStruct('Data', [], properties), 1)}
-
-  struct Errors {
-    let json: Any
+  required init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    self.data = try container.decode(Data.self, forKey: .data)
+    self.errors = try container.decodeIfPresent(Errors.self, forKey: .errors)
+  }
 
-    init?(json: Any) {
-      guard let errors = json as? [String: Any] else {
-        return nil
-      }
+${indentMultiline(printStruct('Data', [], properties), 1)}
 
-      self.json = errors
+  struct Errors: Decodable {
+    init(from decoder: Decoder) throws {
+      // TODO
```

Comment:
Need to define the structure of `errors` (basically `Decodable` doesn't not support decoding an arbitrary dictionary).